### DISCUSSION TITLE
The connection cannot be released

### DISCUSSION THREAD

**Discussion by ethink** [duckdb/duckdb#3671] [General]
Establish a connection query for several times in succession and report an error
The code is as follows：

public static void main(String[] args) {

	execute( "select * from data2");
	
	execute( "select * from data2");
	
}
public static void execute(String sql) {
try
{
long t1=System.currentTimeMillis();
Properties ro_prop = new Properties();
ro_prop.setProperty("duckdb.read_only", "false");
DuckDBConnection conn =(DuckDBConnection) DriverManager.getConnection("jdbc:duckdb:C:\test\duckdb\v3_ptest100_5.db");
Statement stmt = conn.createStatement();
stmt.execute(sql);
stmt.close();
conn.close();
long t2=System.currentTimeMillis();

	      System.out.println("time：" + ( t2-t1)+"ms,sql："+sql);
	    }catch(Exception e) {
	    	e.printStackTrace();
	    }
  }
Error reporting log：
time：450ms,sql：select * from data2
java.sql.SQLException: IO Error: Cannot open file "C:\test\duckdb\v3_ptest100_5.db": Áíһ¸ö³ÌÐòÕýÔÚʹÓôËÎļþ£¬½ø³ÌÎ
at org.duckdb.DuckDBNative.duckdb_jdbc_startup(Native Method)
at org.duckdb.DuckDBDatabase.(DuckDBDatabase.java:22)
at org.duckdb.DuckDBDriver.connect(DuckDBDriver.java:35)
at java.sql.DriverManager.getConnection(Unknown Source)
at java.sql.DriverManager.getConnection(Unknown Source)

**Comment by forzpan** [comment]
the 'close()' method in DuckDBConnection.java just set 'db = null' .
you should change it to 'db.shutdown()',then it worked.
